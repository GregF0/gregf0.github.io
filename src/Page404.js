import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { useTheme } from '@material-ui/core'
import MatBox from '@material-ui/core/Box'
import Typography from '@material-ui/core/Typography'
import illustration from './assets/404.svg'

export const Page404 = () => {
  const theme = useTheme()

  return (
    <Box
      display="flex"
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
      alignSelf="center"
      width="90%"
      maxWidth="600px"
    >
      <img src={illustration} alt="Page not found" />
      <Typography align="center">
        How did you get here?! You should{' '}
        <LinkStyled to="/timer" color={theme.palette.secondary.main}>
          go back to work
        </LinkStyled>
        !
      </Typography>
    </Box>
  )
}

const Box = styled(MatBox)`
  position: absolute;
  top: 40%;
  left: 50%;
  transform: translate(-50%, -40%);
`

const LinkStyled = styled(Link)`
  text-decoration: none;
  color: ${({ color }) => color};
`
