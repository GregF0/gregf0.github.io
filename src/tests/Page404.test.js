import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { createShallow } from '@material-ui/core/test-utils'
import { Page404 } from '../Page404'

describe('<Page404 />', () => {
  let shallow

  beforeAll(() => {
    shallow = createShallow({ untilSelector: Page404 })
  })

  test('should render Page 404 correctly', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <Page404 />
      </BrowserRouter>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
