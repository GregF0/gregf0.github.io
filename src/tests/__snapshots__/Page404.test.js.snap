// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page404 /> should render Page 404 correctly 1`] = `
<Styled(Styled(MuiBox))
  alignItems="center"
  alignSelf="center"
  display="flex"
  flexDirection="column"
  justifyContent="center"
  maxWidth="600px"
  width="90%"
>
  <img
    alt="Page not found"
    src="404.svg"
  />
  <WithStyles(ForwardRef(Typography))
    align="center"
  >
    How did you get here?! You should
     
    <Styled(Link)
      color="#f50057"
      to="/timer"
    >
      go back to work
    </Styled(Link)>
    !
  </WithStyles(ForwardRef(Typography))>
</Styled(Styled(MuiBox))>
`;
